engine/src/main/java/org/camunda/bpm/engine/impl/db/entitymanager/DbEntityManager.java

```diff
@@ -178,8 +179,19 @@ public List selectListWithRawParameter(String statement, Object parameter, int f
     if(firstResult == -1 ||  maxResults==-1) {
       return Collections.EMPTY_LIST;
     }
-    List loadedObjects = persistenceSession.selectList(statement, parameter);
-    return filterLoadedObjects(loadedObjects);
+    try {
+
+      List loadedObjects = persistenceSession.selectList(statement, parameter);
+      return filterLoadedObjects(loadedObjects);
+    } catch (Exception e) {
+
+      if (DbSqlSession.isCrdbConcurrencyConflict(e)) {
+        // on CRDB, selects might fail with an OLE and the transaction must be retried.
+        throw LOG.crdbTransactionRetryExceptionOnSelect(e);
+      }
```

Comment:
This is one out of multiple SELECT methods in this class, but only here we have this treatment. We should implement it everywhere or nowhere. If we go with everywhere, have a look at the latest changes on master and the exception wrapping that has been introduced around all mybatis interactions. In the end, make sure that these exceptions actually bubble up to the command and are not wrapped in more generic ProcessEngineExceptions (or else the user or retry interceptor won't be able to act on them).

Reply:
I removed this, as this impacted Exclusive Message Correlation, which we concluded that is not supported on CRDB. Ticket CAM-12490 tracks the issue.